lib/rubocop/cop/style/redundant_regexp_character_class.rb

```diff
@@ -65,19 +48,54 @@ def on_regexp(node)
         private
 
         def each_redundant_character_class(node)
-          pattern_source(node).scan(PATTERN) do
-            yield match_range(node.loc.begin.end, Regexp.last_match)
+          each_single_element_character_class(node) do |char_class|
+            next unless redundant_single_element_character_class?(node, char_class)
+
+            yield node.loc.begin.adjust(begin_pos: 1 + char_class.ts, end_pos: char_class.te)
+          end
+        end
+
+        def each_single_element_character_class(node)
+          Regexp::Parser.parse(pattern_source(node)).each_expression do |expr|
```

Comment:
I think you should use `node.parsed_tree` here...

Reply:
Unfortunately `parsed_tree` [doesn't support ](https://github.com/rubocop-hq/rubocop/blob/69700a544bcb15079f177ce5df4f808a61f1949f/lib/rubocop/ext/regexp_node.rb#L19-L20)patterns containing an interpolation. @marcandre I see you added that method - do you recall why you return `nil` for patterns containing interpolations? As it stands, this diff causes a bunch of spec failures:
```diff
diff --git a/lib/rubocop/cop/style/redundant_regexp_character_class.rb b/lib/rubocop/cop/style/redundant_regexp_character_class.rb
index 214a84743..7a3251994 100644
--- a/lib/rubocop/cop/style/redundant_regexp_character_class.rb
+++ b/lib/rubocop/cop/style/redundant_regexp_character_class.rb
@@ -54,7 +54,8 @@ module RuboCop
         end
 
         def each_single_element_character_class(node)
-          Regexp::Parser.parse(pattern_source(node)).each_expression do |expr|
+          # Regexp::Parser.parse(pattern_source(node)).each_expression do |expr|
+          node.parsed_tree.each_expression do |expr|
             next if expr.type != :set || expr.expressions.size != 1
             next if expr.negative? || %i[posixclass set].include?(expr.expressions.first.type)
```
mostly of the form `NoMethodError: undefined method `each_expression' for nil:NilClass`.

N.B. I introduced `pattern_source` (which replaces interpolation and comments with spaces) to remove (as far as the cops are concerned) the "effect" of the interpolation, whilst preserving the indentation/width of the interpolation in the pattern

Reply:
Perhaps we could incorporate the "interpolation-blanking" into `parsed_tree`, or some other mechanism of supporting interpolation?

Reply:
I like the idea of blanking the interpolations, makes sense and could be quite useful.
Maybe `parsed_tree` could take a keyword argument `:interpolation` with value `:ignore` or `:blank`?

Reply:
I've had a go at this - please let me know what you think @marcandre!

Reply:
That's awesome.
Ideally, the changes to `parsed_tree` would be in a different PR, and with at least a test.
It's not clear to me that replacing the interpolations with spaces has no effect, I thought you were going to simply delete them...
I also don't know why the treatment of the comments.